### PR TITLE
Increase pod validation timeout, and improve failure message

### LINKED ISSUE #694: Increase validate-pod timeout
I feel a timeout of two minutes is a little short (see below).

[https://github.com/apprenda/kismatic/blob/master/ansible/roles/validate-pod/tasks/validate-pod.yaml#L7-L8](https://github.com/apprenda/kismatic/blob/master/ansible/roles/validate-pod/tasks/validate-pod.yaml#L7-L8)

I am wondering if we should increase this to no longer than five minutes to allow for network connectivity issues.

@alexbrand thoughts?

### PR DESCRIPTION
**Pull request by alexbrand**
Fixes #694 

We have seen various instances of pod validation failing because the API server pod takes longer than the original timeout. We have also seen this with the DNS pod.

Also improved the UX when one of these failures happen. This not only improves error messages, but also diagnosability during integration tests:

* When the pod doesn't have the expected version:
```
Validating==========================================================================
Reading installation plan file "kismatic-cluster.yaml"                          [OK]
Validating installation plan file                                               [OK]
Validating SSH connectivity to nodes                                            [OK]

Running Task========================================================================
Validate Kubernetes Control Plane is Running                                    [ERROR]
- Task: fail if pod 'kube-apiserver-ip-10-0-3-130' does not have the desired version
  ip-10-0-3-130: The pod 'kube-apiserver-ip-10-0-3-130' is expected to have version '1.5.0', but it has version '1.5.0-dirty'.[ERROR]
```

* When a pod takes longer than expected to enter the ready state:
```

Validating==========================================================================
Reading installation plan file "kismatic-cluster.yaml"                          [OK]
Validating installation plan file                                               [OK]
Validating SSH connectivity to nodes                                            [OK]

Running Task========================================================================
Validate Kubernetes Control Plane is Running                                    [ERROR]
- Task: fail if kube-apiserver-ip-10-0-3-130 pod is not running
  ip-10-0-3-130: Waited for pod 'kube-apiserver-ip-10-0-3-130' to be running, but it did not start up in time.

The pod's latest logs may indicate why it failed to start up:

I0801 22:59:05.333632       1 wrap.go:42] GET /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.720274ms) 200 [[kube-controller-manager/v1.7.2 (linux/amd64) kubernetes/922a86c/leader-election] 54.211.237.33:33332]
I0801 22:59:05.340609       1 wrap.go:42] PUT /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (2.712948ms) 200 [[kube-controller-manager/v1.7.2 (linux/amd64) kubernetes/922a86c/leader-election] 54.211.237.33:33332]
I0801 22:59:05.874497       1 wrap.go:42] GET /api/v1/watch/services?resourceVersion=4267&timeoutSeconds=568: (9m28.003605551s) 200 [[nginx-ingress-controller/v0.0.0 (linux/amd64) kubernetes/$Format] 10.0.3.41:34094]
I0801 22:59:05.878363       1 rbac.go:114] RBAC DENY: user "system:serviceaccount:kube-system:default" groups ["system:serviceaccounts" "system:serviceaccounts:kube-system" "system:authenticated"] cannot "watch" resource "services" cluster-wide
I0801 22:59:05.878610       1 rest.go:349] Starting watch for /api/v1/watch/services, rv=4267 labels= fields= timeout=6m12s
I0801 22:59:06.695118       1 wrap.go:42] GET /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (1.807858ms) 200 [[kube-scheduler/v1.7.2 (linux/amd64) kubernetes/922a86c/leader-election] 54.211.237.33:33318]
I0801 22:59:06.698910       1 wrap.go:42] PUT /api/v1/namespaces/kube-system/endpoints/kube-scheduler: (2.351219ms) 200 [[kube-scheduler/v1.7.2 (linux/amd64) kubernetes/922a86c/leader-election] 54.211.237.33:33318]
I0801 22:59:06.936095       1 wrap.go:42] GET /api/v1/watch/pods?resourceVersion=4565: (10.010573652s) 200 [[python-requests/2.13.0] 54.159.140.140:55516]
I0801 22:59:06.949559       1 rbac.go:114] RBAC DENY: user "system:serviceaccount:kube-system:default" groups ["system:serviceaccounts" "system:serviceaccounts:kube-system" "system:authenticated"] cannot "watch" resource "pods" cluster-wide
I0801 22:59:06.949865       1 rest.go:349] Starting watch for /api/v1/watch/pods, rv=4565 labels= fields= timeout=54m5.954863982s
I0801 22:59:07.344904       1 wrap.go:42] GET /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.840436ms) 200 [[kube-controller-manager/v1.7.2 (linux/amd64) kubernetes/922a86c/leader-election] 54.211.237.33:33332]
I0801 22:59:07.348377       1 wrap.go:42] PUT /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (2.221147ms) 200 [[kube-controller-manager/v1.7.2 (linux/amd64) kubernetes/922a86c/leader-election] 54.211.237.33:33332]
I0801 22:59:07.432024       1 wrap.go:42] GET /api/v1/nodes: (4.497246ms) 200 [[kube-controller-manager/v1.7.2 (linux/amd64) kubernetes/922a86c/system:serviceaccount:kube-system:pod-garbage-collector] 54.211.237.33:33350]
I0801 22:59:07.541130       1 wrap.go:42] GET /apis/extensions/v1beta1/ingresses?resourceVersion=4267&timeoutSeconds=569&watch=true: (9m29.000603971s) 200 [[kube-controller-manager/v1.7.2 (linux/amd64) kubernetes/922a86c/shared-informers] 54.211.237.33:33332]
I0801 22:59:07.542505       1 rest.go:349] Starting watch for /apis/extensions/v1beta1/ingresses, rv=4267 labels= fields= timeout=5m47s
                                                                                [ERROR]
error running playbook: error running ansible: exit status 2
```

* When DNS fails to start up
```
Validating==========================================================================
Reading installation plan file "kismatic-cluster.yaml"                          [OK]
Validating installation plan file                                               [OK]
Validating SSH connectivity to nodes                                            [OK]

Running Task========================================================================
Start Kubernetes DNS                                                            [ERROR]
- Task: fail if DNS pods are not ready
  ip-10-0-3-130: Waited for all DNS pods to be ready, but they took longer than 5 minutes to be in the ready state.

The pod's latest logs may indicate why it failed to start up:

I0801 21:38:55.241950       1 server.go:113] FLAG: --nameservers=""
I0801 21:38:55.241953       1 server.go:113] FLAG: --stderrthreshold="2"
I0801 21:38:55.241955       1 server.go:113] FLAG: --v="2"
I0801 21:38:55.241959       1 server.go:113] FLAG: --version="false"
I0801 21:38:55.241973       1 server.go:113] FLAG: --vmodule=""
I0801 21:38:55.242043       1 server.go:176] Starting SkyDNS server (0.0.0.0:10053)
I0801 21:38:55.242215       1 server.go:200] Skydns metrics not enabled
I0801 21:38:55.242223       1 dns.go:147] Starting endpointsController
I0801 21:38:55.242226       1 dns.go:150] Starting serviceController
I0801 21:38:55.243359       1 logs.go:41] skydns: ready for queries on cluster.local. for tcp://0.0.0.0:10053 [rcache 0]
I0801 21:38:55.243389       1 logs.go:41] skydns: ready for queries on cluster.local. for udp://0.0.0.0:10053 [rcache 0]
I0801 21:38:55.742424       1 dns.go:171] Initialized services and endpoints from apiserver
I0801 21:38:55.742439       1 server.go:129] Setting up Healthz Handler (/readiness)
I0801 21:38:55.742444       1 server.go:134] Setting up cache handler (/cache)
I0801 21:38:55.742448       1 server.go:120] Status HTTP port 8081
                                                                                [ERROR]
error running playbook: error running ansible: exit status 2
```

### CODE REVIEW

File: ansible/roles/validate-pod/tasks/validate-pod.yaml [hunk 1]
```diff
@@ -1,33 +1,43 @@
...
     command: kubectl get pods {{ item }} --namespace kube-system -o jsonpath='{.metadata.annotations.kismatic/version}' --kubeconfig {{ kubernetes_kubeconfig_path }}
     register: podVersion
     until: podVersion.stdout == kismatic_short_version
-    retries: 20
-    delay: 6
+    retries: 60
+    delay: 5
+    failed_when: false # We don't want this task to actually fail (We catch the failure with a custom msg in the next task)
+
+  - name: fail if pod '{{ item }}' does not have the desired version
```
Comment (alexbrand) [inline ansible/roles/validate-pod/tasks/validate-pod.yaml]: I actually think we should see if there was anything in stderr and print it as well.

Comment (alexbrand) [inline ansible/roles/validate-pod/tasks/validate-pod.yaml]: This is done. @dkoshkin PTAL.

**Comment by dkoshkin** [comment]
Can we also increase the timeout for ingress, I've seen that fail before

**Comment by alexbrand** [comment]
Definitely!

**Comment by dkoshkin** [comment]
LGTM!

**Pull request merged**